tls/s2n_ktls_io.c

```diff
@@ -319,7 +318,56 @@ ssize_t s2n_ktls_sendv_with_offset(struct s2n_connection *conn, const struct iov
     }
 
     size_t bytes_written = 0;
-    POSIX_GUARD_RESULT(s2n_ktls_sendmsg(conn, TLS_APPLICATION_DATA, bufs, count,
-            blocked, &bytes_written));
+    POSIX_GUARD_RESULT(s2n_ktls_sendmsg(conn->send_io_context, TLS_APPLICATION_DATA,
+            bufs, count, blocked, &bytes_written));
+    return bytes_written;
+}
+
+int s2n_ktls_send_cb(void *io_context, const uint8_t *buf, uint32_t len)
+{
+    /* For now, all control records are assumed to be alerts.
+     * We can set the record_type on the io_context in the future.
+     */
+    const uint8_t record_type = TLS_ALERT;
+
+    const struct iovec iov = {
+        .iov_base = (void *) (uintptr_t) buf,
+        .iov_len = len,
+    };
+    s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+    size_t bytes_written = 0;
+
+    s2n_result result = s2n_ktls_sendmsg(io_context, record_type, &iov, 1,
+            &blocked, &bytes_written);
+
+    POSIX_GUARD_RESULT(result);
```

Comment:
Why store the result if you're just going to POSIX_GUARD it on the next line?

Reply:
Nice catch, I just forgot to clean that up after a refactor :) I'll fix it.